src/java/genomicsdb/VCF2TileDB.java

```diff
@@ -43,14 +61,289 @@
+     */
+    private class SilentByteBufferStream extends OutputStream
+    {
+        private byte mBuffer[] = null;
+        private long mNumValidBytes = 0;
+        private long mMarker = 0;
+        private boolean mOverflow = false;
+        
+        public SilentByteBufferStream()
+        {
+            mBuffer = new byte[(int)mDefaultBufferCapacity];
+        }
+
+        public SilentByteBufferStream(final long capacity)
+        {
+            mBuffer = new byte[(int)capacity];
+        }
+
+        @Override
+        public void close() throws IOException  //does nothing
+        {
+        }
+
+        @Override
+        public void flush() throws IOException  //does nothing
+        {
+        }
+
+        @Override
+        public void write(byte[] b, int off, int len) throws IOException
+        {
+            if(mOverflow)
+                return;
+            if(len+mNumValidBytes > mBuffer.length)
+                mOverflow = true;
+            else
+            {
+                System.arraycopy(b, off, mBuffer, (int)mNumValidBytes, len);
+                mNumValidBytes += len;
+            }
+        }
+
+        @Override
+        public void write(byte[] b) throws IOException
+        {
+            write(b, 0, b.length);
+        }
+
+        @Override
+        public void write(int b) throws IOException
+        {
+            if(mOverflow)
+                return;
+            if(mNumValidBytes+1>mBuffer.length)
+                mOverflow = true;
+            else
+            {
+                mBuffer[(int)mNumValidBytes] = (byte)b;
+                ++mNumValidBytes;
+            }
+        }
+
+        public int size()
+        {
+            return mBuffer.length;
+        }
+
+        public void resize(final long newSize)
+        {
+            byte tmp[] = new byte[(int)newSize];
+            System.arraycopy(mBuffer, 0, tmp, 0, mBuffer.length);
+            mBuffer = tmp; //hopefully Java GC does its job
+        }
+
+        public boolean overflow()
+        {
+            return mOverflow;
+        }
+
+        public void setOverflow(final boolean value)
+        {
+            mOverflow = value;
+        }
+
+        public long getNumValidBytes()
+        {
+            return mNumValidBytes;
+        }
+
+        public void setNumValidBytes(final long value)
+        {
+            mNumValidBytes = value;
+        }
+
+        public void setMarker(final long value)
+        {
+            mMarker = value;
+        }
+
+        public long getMarker()
+        {
+            return mMarker;
+        }
+
+        public byte[] getBuffer()
+        {
+            return mBuffer;
+        }
+    }
 
     /**
-     * JNI function that writes to TileDB
+     * Utility class wrapping a stream and a VariantContextWriter for a given stream
+     * Each GenomicsDB import stream consists of a buffer stream and a writer object
+     */
+    private class GenomicsDBImporterStreamWrapper
```

Comment:
minor comment: It is better to create separate Java files for each class. Code looks cleaner.